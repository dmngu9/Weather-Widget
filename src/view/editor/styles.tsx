import styled from 'styled-components';

import * as fonts from 'utils/fonts';

export const Form = styled.form`
    width: 360px;
    padding: 24px;
`;

export const FieldSet = styled.fieldset`
    margin: 0 0 16px;
    padding: 0;
    border: none;
`;

export const Legend = styled.legend`
    padding: 0;
    margin-bottom: 8px;
    font-size: ${fonts.medium}px;
`;

export const FieldText = styled.input`
    height: 36px;
    width: 100%;
    border-radius: 4px;
    border: solid 1px #ccc;
    padding: 16px;
    box-sizing: border-box;
    outline: none;
    font-size: ${fonts.small}px;

    &:focus {
        border-color: dodgerblue;
        box-shadow: 0 0 8px 0 dodgerblue;
    }

    &::placeholder {
        font-size: ${fonts.small}px;
    }
`;

export const RadioLabel = styled.label`
    display: inline-block;
    cursor: pointer;
    font-size: ${fonts.small}px;
    user-select: none;
    padding-left: 24px;
    position: relative;

    &:hover > span {
        background-color: #ccc;
    }
`;

export const RadioButton = styled.input`
    display: none;

    &:checked + span {
        background-color: dodgerblue;
    }

    &:checked + span::after {
        content: '';
        height: 8px;
        width: 8px;
        background-color: white;
        border-radius: 50%;
        position: absolute;
        left: 50%;
        top: 50%;
        transform: translate(-50%, -50%);
    }
`;

export const Circle = styled.span`
    display: inline-block;
    height: 16px;
    width: 16px;
    background-color: #fff;
    position: absolute;
    left: 0;
    top: 0;
    border-radius: 50%;
`;

export const RadioGroup = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
`;
