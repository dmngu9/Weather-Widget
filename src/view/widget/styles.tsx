import styled from 'styled-components';

import * as fonts from 'utils/fonts';

export const Container = styled.div`
    width: 320px;
    padding: 24px;
    background-color: white;
    box-sizing: border-box;
    border-radius: 4px;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
    transition: transform 0.3s ease-in-out;

    &:hover {
        transform: scale(1.05);
    }
`;

export const Title = styled.h3`
    margin: 0 0 24px;
    text-transform: uppercase;
    word-break: break-word;
    hyphens: auto;
`;

export const Content = styled.div`
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    flex-wrap: nowrap;
`;

export const IconContainer = styled.div`
    height: 120px;
    width: 120px;
    flex-shrink: 0;
    object-fit: fill;
    margin-right: 24px;
`;

export const Icon = styled.img`
    height: 100%;
    width: 100%;
`;

export const Info = styled.div`
    word-break: break-word;
    hyphens: auto;
`;

export const City = styled.p`
    font-size: ${fonts.medium}px;
    margin: 0 0 8px;
`;

export const Temperature = styled.p`
    font-size: ${fonts.xlarge}px;
    font-weight: bold;
    margin: 0 0 8px;
`;

export const Wind = styled.p`
    font-size: ${fonts.small}px;
    margin: 0;
`;
