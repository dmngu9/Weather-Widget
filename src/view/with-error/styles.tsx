import styled from 'styled-components';

import * as fonts from 'utils/fonts';

export const Error = styled.p`
    font-size: ${fonts.large};
    color: red;
`;
