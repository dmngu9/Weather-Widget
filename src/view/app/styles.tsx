import styled from 'styled-components';

import { Editor } from 'view/editor';
import { Widget } from 'view/widget';
import { media } from 'utils/breakpoints';

export const Container = styled.div`
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
`;

export const StyledEditor = styled(Editor)`
    ${media.tablet`
		border: none;
		margin-bottom: 32px;
	`};

    border-right: 1px solid #ccc;
    margin-right: 24px;
    flex-shrink: 0;
`;

export const StyledWidget = styled(Widget)`
    flex-shrink: 0;
`;
