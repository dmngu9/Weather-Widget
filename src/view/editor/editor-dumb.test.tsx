import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Editor from './editor-dumb';
import { FieldText, RadioButton } from './styles';

describe('<Editor />', () => {
    const callback = jest.fn();
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        callback.mockReset();
        wrapper = shallow(<Editor onChange={callback} unit="metric" wind="on" />);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('should call onChange when form field change', () => {
        it('should callback when title input text field', () => {
            wrapper.find(FieldText).simulate('change', { target: { value: 'hello' } });

            expect(callback).toHaveBeenCalledWith({ target: { value: 'hello' } });
        });

        it('should callback when unit radio button is checked', () => {
            wrapper
                .find(RadioButton)
                .filterWhere(radio => radio.prop('value') === 'imperial')
                .simulate('click', { target: { value: 'imperial' } });

            expect(callback).toHaveBeenCalledWith({ target: { value: 'imperial' } });
        });

        it('should callback when wind radio button is checked', () => {
            wrapper
                .find(RadioButton)
                .filterWhere(radio => radio.prop('value') === 'off')
                .simulate('click', { target: { value: 'off' } });

            expect(callback).toHaveBeenCalledWith({ target: { value: 'off' } });
        });
    });
});
