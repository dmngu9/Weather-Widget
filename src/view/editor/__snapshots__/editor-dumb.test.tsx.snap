// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Editor /> should render correctly 1`] = `
<styled.form>
  <styled.fieldset>
    <styled.legend>
      Title
    </styled.legend>
    <styled.input
      maxLength={50}
      name="title"
      onChange={[MockFunction]}
      placeholder="Title of widget"
      type="text"
    />
  </styled.fieldset>
  <styled.fieldset>
    <styled.legend>
      Unit
    </styled.legend>
    <styled.div>
      <styled.label>
        Metric (°C, m/s)
        <styled.input
          defaultChecked={true}
          name="unit"
          onClick={[MockFunction]}
          type="radio"
          value="metric"
        />
        <styled.span />
      </styled.label>
      <styled.label>
        Imperial (°F, mph)
        <styled.input
          defaultChecked={false}
          name="unit"
          onClick={[MockFunction]}
          type="radio"
          value="imperial"
        />
        <styled.span />
      </styled.label>
    </styled.div>
  </styled.fieldset>
  <styled.fieldset>
    <styled.legend>
      Wind
    </styled.legend>
    <styled.div>
      <styled.label>
        On
        <styled.input
          defaultChecked={true}
          name="wind"
          onClick={[MockFunction]}
          type="radio"
          value="on"
        />
        <styled.span />
      </styled.label>
      <styled.label>
        Off
        <styled.input
          defaultChecked={false}
          name="wind"
          onClick={[MockFunction]}
          type="radio"
          value="off"
        />
        <styled.span />
      </styled.label>
    </styled.div>
  </styled.fieldset>
</styled.form>
`;
